ofrak_core/ofrak/service/data_service.py

```diff
@@ -134,6 +146,33 @@ async def delete_models(self, data_ids: Iterable[DataId]) -> None:
             root.delete_mapped_model(model)
             del self._model_store[model.id]
 
+    async def search(self, data_id, query, start=None, end=None, max_matches=None):
```

Comment:
Type annotation on return value?

Reply:
The typing is all done in the `DataServiceInterface` because it requires 2 type stubs via overloading.